From c2316b1f727a218e9a9dadbc9b2ef95097af35be Mon Sep 17 00:00:00 2001
From: mx3L <mx3ldev@gmail.com>
Date: Sun, 27 Mar 2016 13:11:12 +0200
Subject: [PATCH 08/11] first flush then change track

If it's done in opposite way, we sometimes loose caps event
in sinks.

If we have audio-tracks with different codecs, say
we have AAC and AC3. AAC is selected and running, we
switch to AC3, no caps event with AAC caps in sink ->
AC3 is written to decoder as AAC -> no sound.
---
 gstplayer/common/tracks.h | 28 ++++++++++------------------
 1 file changed, 10 insertions(+), 18 deletions(-)

diff --git a/gstplayer/common/tracks.h b/gstplayer/common/tracks.h
index 9faf494..4d766a5 100644
--- a/gstplayer/common/tracks.h
+++ b/gstplayer/common/tracks.h
@@ -139,17 +139,13 @@ static int SelectSubtitleTrack(unsigned int i)
                 ppos = 0;
             }
         }
-
-        ret = SelectSubtitleStream(i);
-        if (!ret)
+        if (validposition)
         {
-            if (validposition)
-            {
-                /* flush */
-                double dpos = ppos/1000.0;
-                backend_seek_absolute(dpos);
-            }
+            /* flush */
+            double dpos = ppos/1000.0;
+            backend_seek_absolute(dpos);
         }
+        ret = SelectSubtitleStream(i);
     }
     else
     {
@@ -190,17 +186,13 @@ static int SelectAudioTrack(unsigned int i)
                 ppos = 0;
             }
         }
-
-        ret = SelectAudioStream(i);
-        if (!ret)
+        if (validposition)
         {
-            if (validposition)
-            {
-                /* flush */
-                double dpos = ppos/1000.0;
-                backend_seek_absolute(dpos);
-            }
+            /* flush */
+            double dpos = ppos/1000.0;
+            backend_seek_absolute(dpos);
         }
+        ret = SelectAudioStream(i);
     }
     else
     {
-- 
1.9.1

